### PR TITLE
Support module unification app in `/src/main`

### PR DESCRIPTION
**Pull request by iezer**
@rwjblue  @kratiahuja FYI this change is need in order to use fastboot with a module unification app.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -77,6 +77,17 @@ module.exports = {
     }
   },
 
+  /**
+     Returns whether this app is using a module unification format.
+     @private
+     @method isModuleUnification
+     @return {Boolean} Whether this app is using a module unification format.
+  */
+  isAppModuleUnification() {
```
Comment (rwjblue) [inline index.js]: Should be using `this.project.isModuleUnificatiob()` (no need for a new helper method AFAICT)

Comment (iezer) [reply index.js]: @rwjblue Good call, I've made that change.,

File: index.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ module.exports = {
     }
 
     if (type === 'app-boot') {
-      return fastbootAppModule(config.modulePrefix, JSON.stringify(config.APP || {}));
+      return fastbootAppModule(config.modulePrefix, JSON.stringify(config.APP || {}), this.project.isModuleUnification());
```
Comment (rwjblue) [inline index.js]: Hmm, we may need to handle the case where the function doesn’t exist (IIRC it only exists on ember-cli master).

Comment (iezer) [reply index.js]: as in `this.project.isModuleUnification && this.project.isModuleUnification()` ?

Comment (iezer) [reply index.js]: or perhaps `(typeof this.project.isModuleUnification === 'function') && this.project.isModuleUnification()`

Comment (iezer) [reply index.js]: fixed.

**Comment by kratiahuja** [comment]
How soon does need to land?

**Comment by rwjblue** [comment]
@kratiahuja - yeah, agreed

**Comment by kratiahuja** [comment]
Ok I will release this first and then release beta.

**Pull request merged**